### ISSUE TITLE
Undefined reference to symbol _ZN4llvm15SmallPtrSetImplD2Ev

### ISSUE THREAD

**Issue by adityaatluri** [imageworks/OpenShadingLanguage#574]
Here is the error stack.
[ 21%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/shadeimage.cpp.o
[ 22%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/backendllvm.cpp.o
[ 24%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/llvm_gen.cpp.o
[ 25%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/llvm_instance.cpp.o
[ 26%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/llvm_util.cpp.o
[ 27%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/__/liboslcomp/ast.cpp.o
[ 28%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/__/liboslcomp/codegen.cpp.o
[ 29%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/__/liboslcomp/oslcomp.cpp.o
[ 31%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/__/liboslcomp/symtab.cpp.o
[ 32%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/__/liboslcomp/typecheck.cpp.o
[ 33%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/__/liboslquery/oslquery.cpp.o
[ 34%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/osogram.cpp.o
[ 35%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/osolex.cpp.o
[ 36%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/oslgram.cpp.o
[ 37%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/osllex.cpp.o
[ 39%] Building CXX object src/liboslexec/CMakeFiles/oslexec.dir/llvm_ops.bc.cpp.o
Linking CXX shared library liboslexec.so
[ 75%] Built target oslexec
Scanning dependencies of target accum_test
[ 77%] Building CXX object src/liboslexec/CMakeFiles/accum_test.dir/accum_test.cpp.o
Linking CXX executable accum_test
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../lib/libLLVMMCJIT.a(MCJIT.o): undefined reference to symbol '_ZN4llvm15SmallPtrSetImplD2Ev'
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../x86_64-linux-gnu/libLLVM-3.4.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [src/liboslexec/accum_test] Error 1
make[1]: *** [src/liboslexec/CMakeFiles/accum_test.dir/all] Error 2
make: *** [all] Error 2

**Comment by adityaatluri** [comment]
I downloaded libLLVM-3.3.so.1 and soft linked as libLLVM-3.4
```
ln -s libLLVM-3.3.so.1 libLLVM-3.4.so
```

**Comment by lgritz** [comment]
That link doesn't look very safe.

I don't recognize those errors, am not sure why it wouldn't work. The "SmallPtrSetImpl" is purely internal to LLVM, it's not something we reference anywhere in OSL. 

The only thing I can think of is that maybe your system has multiple LLVM implementations installed, and you were getting headers from one, but trying to link against a different version of the library?

**Comment by adityaatluri** [comment]
Yes. OSL does not reference that function. But, I think certain LLVM APIs used in OSL needs it. 

Well, yes. I was getting the error when my system has LLVM 3.4 (default on Ubuntu 14.04). The ```cmake``` ran well but, when I used ```make```, it ran into the above error. So, I downloaded libLLVM-3.4 debian package and installed it. It still threw the same error. I guessed to using libLLVM-3.3.so would solve the problem and it did. I think the problem is specific to my system. It seems to work for me.
